### PR TITLE
integration_tests: add test for LP: #1898997

### PR DESCRIPTION
**Pull request by OddBloke**
## Proposed Commit Message
```
integration_tests: add test for LP: #1898997

This introduces the `lxd_config_dict` mark, used to specify a
free-form configuration dict to LXD for tests which only run there; and
the `not_xenial` and `not_bionic` marks, used to skip tests on
xenial/bionic via a basic release skipping mechanism.

(The `lxd_config_dict` change further complicated `_client`, so a minor
refactoring is applied.)
```

## Additional Context

This is WIP because it needs https://github.com/canonical/pycloudlib/pull/66 to land for network config to LXD VMs to work.

## Test Steps

Run `CLOUD_INIT_PLATFORM=lxd_vm pytest --log-cli-level=INFO tests/integration_tests/bugs/test_lp1898997.py` to see a failure.

Run `CLOUD_INIT_PLATFORM=lxd_vm CLOUD_INIT_CLOUD_INIT_SOURCE=PROPOSED pytest --log-cli-level=INFO tests/integration_tests/bugs/test_lp1898997.py` to see a pass.

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

File: tests/integration_tests/bugs/test_lp1898997.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+@pytest.mark.lxd_config_dict({
+    "user.network_config": NETWORK_CONFIG,
+    "volatile.eth0.hwaddr": MAC_ADDRESS,
+})
+@pytest.mark.lxd_vm
+@pytest.mark.sru_2020_11
+@pytest.mark.ubuntu
+class TestInterfaceListingWithOpenvSwitch:
+    def test_ovs_member_interfaces_not_excluded(self, client):
```
Comment (OddBloke) [inline tests/integration_tests/bugs/test_lp1898997.py:44]: After the changes to pycloudlib, this test is now not passing/failing as expected; I need to debug the issue.

Comment (OddBloke) [reply tests/integration_tests/bugs/test_lp1898997.py:46]: This was a simple typo, thankfully!

**Comment by OddBloke** [comment]
This is now ready for review; it still needs https://github.com/canonical/pycloudlib/pull/66 to land before it can land (so I've left the `wip` tag) but I wouldn't anticipate any review changes to that PR would lead to changes here (because, essentially, this is exercising the existing pycloudlib API which was found to have an internal bug).

**Comment by OddBloke** [comment]
This is now ready for review (and subsequent landing, once our Travis issues are sorted).

File: tests/integration_tests/bugs/test_lp1898997.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          macaddress: {}
+version: 2
+""".format(MAC_ADDRESS)
+
+
+@pytest.mark.lxd_config_dict({
+    "user.network-config": NETWORK_CONFIG,
+    "volatile.eth0.hwaddr": MAC_ADDRESS,
+})
+@pytest.mark.lxd_vm
```
Comment (TheRealFalcon) [inline tests/integration_tests/bugs/test_lp1898997.py:40]: lxd_container should work here too, right?

**Review by TheRealFalcon — APPROVED** [review]
Some minor questions/nits, but nothing that should prevent merging. Thanks for all the LXD yak shaving!

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Jk...just remembered something. You should update the commit hash of pycloudlib to the latest commit in integration-requirements.txt. I also forgot to do that in my last PR.

File: tests/integration_tests/conftest.py [hunk 3]
```diff
@@ -182,14 +187,20 @@ def _client(request, fixture_utils, session_cloud):
     Launch the dynamic IntegrationClient instance using any provided
     userdata, yield to the test, then cleanup
     """
-    user_data = fixture_utils.closest_marker_first_arg_or(
-        request, 'user_data', None)
-    name = fixture_utils.closest_marker_first_arg_or(
-        request, 'instance_name', None
-    )
+    def getter(key):
```
[the remark this answers is not in the document]
Comment (OddBloke) [reply tests/integration_tests/conftest.py:190]: Yep!

File: tox.ini [hunk 4]
```diff
@@ -173,8 +173,11 @@ markers =
     gce: test will only run on GCE platform
     azure: test will only run on Azure platform
     oci: test will only run on OCI platform
+    lxd_config_dict: set the config_dict passed on LXD instance creation
```
[the remark this answers is not in the document]
Comment (OddBloke) [reply tox.ini:176]: Yep, agreed. This started off as a generic `network_config` mark, and got refactored from there so it didn't occur to me to use `session_cloud` directly.

File: tests/integration_tests/bugs/test_lp1898997.py (hunk 2 above)
Comment (OddBloke) [reply tests/integration_tests/bugs/test_lp1898997.py:40]: I'll double-check, but IIRC OVS pokes at the kernel in a way that isn't permitted from (non-privileged) containers.

**Comment by OddBloke** [comment]
> Jk...just remembered something. You should update the commit hash of pycloudlib to the latest commit in integration-requirements.txt. I also forgot to do that in my last PR.

Ooh, good catch. I wonder if we should update the test steps to instruct people to manually run the integration tests using `tox`; that would catch such issues, I think.

File: tests/integration_tests/bugs/test_lp1898997.py (hunk 2 above)
Comment (OddBloke) [reply tests/integration_tests/bugs/test_lp1898997.py:40]: Yep, they fail to come back up once `openvswitch` is installed:

```
$ lxc console --show-log intimate-mullet

Console log:

         Starting Cleanup of Temporary Directories...
[  OK  ] Finished Cleanup of Temporary Directories.
[***   ] A start job is running for OpenVSwi… for ovs-br (19min 44s / no limit)

```

**Review by TheRealFalcon — APPROVED** [review]

**Pull request merged**